aws-fargate/miniwdl_aws_fargate.py

```diff
@@ -0,0 +1,208 @@
+                    elif cls.efs_mountpoint != partition.mountpoint:
+                        msg = "Expected filesystem mount {} to match configured EFS mountpoint {}"
+                        raise RuntimeError(msg.format(partition, cls.efs_mountpoint))
+                    return
+        else:
+            if cls.efs_id is None:
+                raise RuntimeError(
+                    'EFS filesystem config is missing. Mount the filesystem and run miniwdl in a subdirectory of the '
+                    'mountpoint or set the filesystem ID with "export MINIWDL__AWS_FARGATE__EFS_ID=fs-12345678 '
+                    'MINIWDL__AWS_FARGATE__EFS_MOUNTPOINT=/mnt/efs" or any other miniwdl config facility. '
+                    'Use "aws efs describe-file-systems" to list filesystems.'
+                )
+            logger.info("Mounting %s at %s", cls.efs_id, cls.efs_mountpoint)
+            fs_url = f"{cls.efs_id}.efs.{ecs.clients.ecs.meta.region_name}.amazonaws.com:/"
+            subprocess.run(["sudo", "mount", "-t", "nfs4", "-o",
+                            "nfsvers=4.1,rsize=1048576,wsize=1048576,hard,timeo=600,retrans=2",
+                            fs_url, cls.efs_mountpoint])
+
+    @classmethod
+    def detect_resource_limits(cls, cfg: config.Loader, logger: logging.Logger) -> Dict[str, int]:
+        return cls._limits
+
+    def poll_task(
+        self, logger: logging.Logger, task_desc, verbose: bool = False
+    ) -> Optional[int]:
+        res = ecs.clients.ecs.describe_tasks(cluster=task_desc["clusterArn"], tasks=[task_desc["taskArn"]])
+        task_desc = res["tasks"][0]
+        if task_desc["lastStatus"] not in self._observed_states:
+            logger.info("Task %s %s", task_desc["taskArn"], task_desc["lastStatus"])
+            self._observed_states.add(task_desc["lastStatus"])
+        if task_desc["lastStatus"] == "STOPPED" and task_desc.get("stopCode") == "TaskFailedToStart":
+            raise Interrupted(task_desc.get("stoppedReason"))
+        return task_desc.get("containers", [{}])[0].get("exitCode")
+
+    def _run(self, logger: logging.Logger, terminating: Callable[[], bool], command: str) -> int:
+        self._observed_states = set()
+        with open(os.path.join(self.host_dir, "command"), "x") as outfile:
+            outfile.write(command)
+
+        image_tag = self.runtime_values.get("docker", "ubuntu:18.04")
+        if ":" not in image_tag:
+            image_tag += ":latest"
+        logger.info("docker image tag %s", image_tag)
+
+        if os.stat(self.host_dir).st_dev != os.stat(self.efs_mountpoint).st_dev:
+            raise RuntimeError(f"miniwdl run directory is outside EFS mountpoint {self.efs_mountpoint}")
+
+        for host_path, container_path in self.input_file_map.items():
+            real_host_path = os.path.realpath(host_path)
+            host_work_path = os.path.join(self.host_dir, "work/_miniwdl_inputs/0", os.path.basename(real_host_path))
```

Comment:
The 0 is liable to differ when there are multiple input files with the same basename (from different host directories) -- the integer's purpose is to de-conflict just that situation.

Reply:
Ah I see, I've updated this to use the correct logic from the base class.